component/prometheus/remotewrite/types.go

```diff
@@ -111,7 +111,7 @@ func (r *QueueOptions) SetToDefault() {
 
 func (r *QueueOptions) toPrometheusType() config.QueueConfig {
 	if r == nil {
-		return config.DefaultQueueConfig
+		return DefaultQueueOptions.toPrometheusType()
```

Comment:
I feel like something like it would be cleaner to create a new `QueueOptions` instead, call `SetToDefault()` and then call and return the result of `toPrometheusType()`. Otherwise we'd make the same error in the future.

Reply:
Can you show me a code snippet? I'm not sure how that's different than this.

Reply:
I meant something like this:
```
q := QueueOptions{}
q.SetToDefault()
return q.toPrometheusType()
```

The difference is that there is only one place in the code where the defaults are applied - in `SetToDefault()`. Having it in more than one place is what caused the original problem. I suppose this the snipped above is a bit ugly and inefficient, but maybe could get better with some refactoring.

Sorry for nitpicking, it's not a strong requirement on my side, just a suggestion 😄  Feel free to ignore if you're busy.